lib/growth_web/live/growth_live.ex

```diff
@@ -10,21 +10,45 @@ defmodule GrowthWeb.GrowthLive do
 
   @impl true
   def handle_event("base_info", %{"child" => child_params}, socket) do
-    case Growth.create_child(child_params) do
+    child_params
+    |> map_keys_to_atom()
+    |> child_transforms()
+    |> Growth.create_child()
+    |> case do
       {:ok, child} ->
         {:noreply, assign(socket, child: child, loading: false)}
+
       {:error, _reason} ->
         {:noreply, assign(socket, loading: false)}
     end
   end
 
   @impl true
-  def handle_event("measure_info", %{"child" => child, "measure" => measure_params}, socket) do
-    case Growth.child_measure(measure_params, child) do
+  def handle_event("measure_info", %{"measure" => measure_params}, socket) do
+    measure_params
+    |> map_keys_to_atom()
+    |> Growth.child_measure(Map.get(socket, :child))
+    |> case do
       {:ok, measure} ->
-        {:noreply, assign(socket, %{child: child, measure: measure}, loading: false)}
+        {:noreply, assign(socket, measure: measure, loading: false)}
+
       {:error, _reason} ->
-        {:noreply, assign(socket, child: child, loading: false)}
+        {:noreply, assign(socket, loading: false)}
     end
   end
+
+  def map_keys_to_atom(attrs) do
+    # NOTE: (jpd) this is kind of a risk, because one can exploit it and exhaust all atoms
+    Enum.into(attrs, %{}, fn {key, value} -> {String.to_atom(key), value} end)
```

Comment:
We should handle this better. Perhaps add `ecto` or another validator/transformer to handle this step.